sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinHelper.scala

```diff
@@ -207,13 +206,13 @@ object StreamingSymmetricHashJoinHelper extends Logging {
    */
   class StateStoreAwareZipPartitionsRDD[A: ClassTag, B: ClassTag, V: ClassTag](
       sc: SparkContext,
-      f: (Iterator[A], Iterator[B]) => Iterator[V],
-      rdd1: RDD[A],
-      rdd2: RDD[B],
+      var f: (Int, Iterator[A], Int, Iterator[B]) => Iterator[V],
+      var rdd1: RDD[A],
+      var rdd2: RDD[B],
       stateInfo: StatefulOperatorStateInfo,
       stateStoreNames: Seq[String],
       @transient private val storeCoordinator: Option[StateStoreCoordinatorRef])
-      extends ZippedPartitionsRDD2[A, B, V](sc, f, rdd1, rdd2) {
```

Comment:
nit: please update the class description ("This is class is a modified version of [[ZippedPartitionsRDD2]].").

Reply:
Actually the description still hold true, as it only changes the function signature from ZippedPartitionsRDD2 (but cannot extend ZippedPartitionsRDD2 due to the change).

Reply:
Hmm, we can just replace the "ZippedPartitionsRDD2" with "ZippedPartitionsBaseRDD" in the class description.

Reply:
Maybe we can describe this as "a variant of ZippedPartitionsRDD2" as it is really slightly different from ZippedPartitionsRDD2 - more alike copy & paste, then modify. WDYT?

Reply:
OK